deployment/utils/gcf-pin-ipfs-hash/index.js

```diff
@@ -0,0 +1,57 @@
+const IpfsClusterAPI = require('./ipfs-cluster-api-service.js');
+const ipfsClusterApiService = new IpfsClusterAPI(
+  process.env.IPFS_CLUSTER_URL,
+  process.env.IPFS_CLUSTER_USERNAME,
+  process.env.IPFS_CLUSTER_PASSWORD
+);
+
+/*
+Example of incoming data:
+{
+  type: "listing",
+  ipfsHash: "hash",
+  rawData: {
+    "title": "",
+    ...
+  }
+}
+*/
+
+exports.pinService = async (event) => {
+  const pubsubMessage = event.data;
+  const data = pubsubMessage
+    ? JSON.parse(Buffer.from(pubsubMessage, 'base64').toString())
+    : null;
+
+  let pinnedHashes = [];
+  let unPinnedHashes = [];
+
+  if (!Object.is(data, null)) {
+    const hashesToPin = parseIncomingData(data);
+    for(let i = 0; i < hashesToPin.length; i++){
+      hashToPin = hashesToPin[i];
+      pinned = await ipfsClusterApiService.pin(hashToPin);
```

Comment:
Should we have a reasonable number of retries before giving up ? Perhaps with some exponential delay.